eclipse-emfcloud/ecore-glsp
# Implement Undo/Redo and Enotation resource handling via modelserver

Closes #72, Closes #19, Closes #82, Closes #90, Closes #26, Closes #20

- Update to latest modelserver and modelserver-theia versions
- Implement Undo/Redo via modelserver (#72)
- Introduce resource handling for enotation file (#82)
  - Modelserver resource set holds both ecore and enotation file
  - Replace direct notation resource changes by modelserver requests
  - Ensure model operations include commands for notation model as well if applicable
  - Add enotation package registration and remove EcorePackageConfiguration (it is configured per default anyway)
- Fix inheritance implementation (#19)
  - Introduce eSuperType concept
  - Extend ENotation model
  - NotationElements may have types
   - Edges may have source and target elements (#90)
   - Keep inheritanceEdges in a separate BiMap in the EcoreModelIndex as it has no semanticProxyElement
- Update debug launch config for frontend
- Add icons for abstract and interface EClasses (visible in tool palette and tree editor)
- Update property view
  - Temporarily clear selection for property view on 'remove' command update
  - Use URI in widget provider to stabilize checks on incremental updates
- Update tree editor
  - Remove unused code for command and tree updates
  - Fix display of EDatatypes in tree nodes
- Update ecore examples
- Fix set of eAttribute type
  - Use correct eReference (ETypedElement_eType) for SetCommand
  - Remove initialization of attribute with EString type
  - Enable 'Change Attributetype' e2eTest
- Update e2etests
  - Update test resources and test cases to current model state
  - Simplify test execution via configuration file
  - Temporarily disable inheritance edge delete test
  - Extend README for optional video recording to investigate test runs
- Add EOperation and occurrences (#26 , #20)
  - Add EOperation support to diagram and tree editor
  - Add occurrences to Class childs in diagram and tree editor
- Simplify usage of starting eclipse products from Jars
  - Remove "isRunning" flag from GLSPLaunchOptions
  - Introduce runtime argument --startFromJar to determine if servers are started or should be started from JARs
  - Update Readmes

Signed-off-by: Nina Doschek <ndoschek@eclipsesource.com>
